src/main/java/com/google/collegeplanner/data/Course.java

```diff
@@ -0,0 +1,222 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.package com.google.collegeplanner.data;
+
+package com.google.collegeplanner.data;
+
+import java.text.ParseException;
+import java.util.GregorianCalendar;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+public class Course {
+  /*
+   * courseId represents the name of the course that
+   * can be found on the course catalog. Example, ENGL101
+   */
+  private String courseId;
+  /*
+   * name represents the full name of the course.
+   * Example, Academic Writing
+   */
+  private String name;
+  /*
+   * semester represents the 4 digit year followed by
+   * the two digit month when the course starts. Example, 202008
+   */
+  private String semester;
+  /*
+   * credits represents the number of credits the course is worth.
+   */
+  private int credits;
+  /*
+   * departmentId represents the 4 character ID of the
+   * department that the class falls into. Example, ENGL
+   */
+  private String departmentId;
+  /*
+   * description represents the description of
+   * the class found on the course catalog.
+   */
+  private String description;
+  /*
+   * gradingMethod represents the list of possible grading
+   * methods students can opt for. Example, regular, pass-fail.
+   */
+  private String[] gradingMethod;
+  /*
+   * corequisites represents the corequisite classes for this
+   * course. Example, BIO106 is the lab for BIO101 and
+   * must be taken together
+   */
+  private String corequisites;
+  /*
+   * prerequisites represents the prerequisite classes for
+   * this course. Example, MATH101, MATH201, and
+   * either MATH105 or MATH115.
+   */
+  private String prerequisites;
+  /*
+   * restrictions represents the restrictions placed on the
+   * registration of this course. Example, This
+   * class is only avaiable to incoming Freshman.
+   */
+  private String restrictions;
+  /*
+   * additionalInfo represents the additional information
+   * for this course found on the course catalog.
+   */
+  private String additionalInfo;
+  /*
+   *creditGrantedFor represents the classes this course
+   * grants credit for. Example, THET285 grants credit for COMM107
+   */
+  private String creditGrantedFor;
+
+  public Course(String courseId, String name, String semester, int credits, String departmentId,
```

Comment:
btw, instead of a ton of constructor arguments, the Java preferred way is the "Builder" pattern (https://dzone.com/articles/design-patterns-the-builder-pattern). We don't need to change this now (because I think this will likely undergo a bit more revision to deal with datastore), but just wanted to let you know this option exists. It will simplify your tests too.